### PR TITLE
/go/libraries/doltcore/sqle: Remove temporary tables root value in favor of session-scoped tables

### PR DESCRIPTION
**Pull request by andrew-wm-arthur**

### CODE REVIEW

File: go/libraries/doltcore/sqle/dsess/session.go [hunk 1]
```diff
@@ -986,23 +960,31 @@ func (sess *Session) AddDB(ctx *sql.Context, dbState InitialDbState) error {
...
+			sess.tmpTables = append(sess.tmpTables[:i], sess.tmpTables[i+1:]...)
+			break
+		}
 	}
+}
 
-	dbState, _, err := sess.LookupDbState(ctx, dbName)
-	if err != nil {
-		return err
+func (sess *Session) GetTemporaryTable(ctx *sql.Context, name string) (sql.Table, bool) {
```
Comment (andrew-wm-arthur) [inline go/libraries/doltcore/sqle/dsess/session.go:976]: According to the MySQL [docs](https://dev.mysql.com/doc/refman/8.0/en/create-temporary-table.html):
"TEMPORARY tables have a very loose relationship with databases (schemas)."

Should we add a database name param and check it?

Comment (andrew-wm-arthur) [reply go/libraries/doltcore/sqle/dsess/session.go:976]: MySQL does, we should too

File: go/libraries/doltcore/sqle/temp_table.go [hunk 2]
```diff
@@ -0,0 +1,410 @@
...
+
+func (t *TempTable) CreateForeignKey(
+	ctx *sql.Context,
+	fkName string,
+	columns []string,
+	referencedTable string,
+	referencedColumns []string,
+	onUpdate, onDelete sql.ForeignKeyReferenceOption,
+) error {
+	return errors.New("temporary tables do not support foreign keys")
```
Comment (zachmu) [inline go/libraries/doltcore/sqle/temp_table.go:243]: This probably belongs in the engine instead, add a TODO?

File: go/libraries/doltcore/sqle/temp_table.go [hunk 3]
```diff
@@ -0,0 +1,410 @@
...
+}
+func (t *TempTable) DropForeignKey(
+	ctx *sql.Context,
+	fkName string,
+) error {
+	return errors.New("temporary tables do not support foreign keys")
```
Comment (zachmu) [inline go/libraries/doltcore/sqle/temp_table.go:249]: Same here

**Review by zachmu — APPROVED** [review]
LGTM

**Comment by andrew-wm-arthur** [comment]
closing and reopening this PR to run CI :/

**Pull request closed without merging**